csvcubed-pmd/csvcubedpmd/urireplacement/urireplacement.py

```diff
@@ -0,0 +1,90 @@
+import click
+import sys
+import os
+
+from chardet.universaldetector import UniversalDetector
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+
+def _file_in_line(line: str) -> bool:
+    """
+    Returns true if line has file:/ in it
+    """
+    return "file:/" in line
+
+
+def _line_replace(line: str, values: tuple[tuple[str, str]]) -> str:
+    """
+    Replaces given value pairs in line, returns result
+    """
+
+    logging.debug(f"Original line: {line}")
+    if type(values[0]) == str:
```

Comment:
Ah, okay, I expect the type should really be `Union[Tuple[str, str], List[Tuple[str, str]]]` which is to say that that `values` is *either* a `Tuple[str, str]` *or* a `List[Tuple[str, str]]`.

In which case, I think it makes sense to check whether `isinstance(values, list)` to differentiate these two conditions.

You could also consider just allowing values to be a `List[Tuple[str, str]]`, in which case the user must always pass in a list of the values, even if there's just one. It would simplify your code a bit.